compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -233,6 +235,18 @@ public Symbol getBCompiledSymbol(BSymbol symbol, String name) {
             return createXMLNamespaceSymbol((BXMLNSSymbol) symbol);
         }
 
+        if (symbol.kind == SymbolKind.RESOURCE_PATH_IDENTIFIER_SEGMENT) {
+            return createNamedPathSegment((BResourcePathSegmentSymbol) symbol);
```

Comment:
Isn't this name inconsistent with `createPathParamSymbol`? Shouldn't we include Symbol?

Isn't the consistent name `createPathNameSymbol`?

Reply:
Good suggestion!